polars/polars-core/src/series/mod.rs

```diff
@@ -511,11 +512,15 @@ impl Series {
     }
 
     /// Get the sum of the Series as a new Series of length 1.
+    /// Returns a Series with a single-zeroed entry if the self is empty.
     ///
     /// If the [`DataType`] is one of `{Int8, UInt8, Int16, UInt16}` the `Series` is
     /// first cast to `Int64` to prevent overflow issues.
     pub fn sum_as_series(&self) -> Series {
         use DataType::*;
+        if self.is_empty() {
+            return Series::new("", [0]).cast(&Int64).unwrap().sum_as_series();
```

Comment:
This return type is not correct. We should not be casting floats, strings or datelike types to `Int64`.

Reply:
My bad -- I added a check for `self.dtype().is_numeric()` and casted the zeroed value to the corresponding type